src/designer/elsa-workflows-studio/src/components/controls/elsa-input-tags/elsa-input-tags-dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import { Component, h, Prop, Event, EventEmitter, State, Watch } from '@stencil/core';
+import { MultiTextDefinition } from '../../../models/domain';
+
+@Component({
+    tag: 'elsa-input-tags-dropdown',
+    styleUrl: 'elsa-input-tags-dropdown.css',
+    shadow: false,
+})
+export class ElsaInputTagsDropdown {
+
+    @Prop() fieldName?: string;
+    @Prop() fieldId?: string;
+    @Prop() placeHolder?: string = 'Add tag';
+    @Prop() values?: Array<string | MultiTextDefinition> = [];
+    @Prop() dropdownValues?: Array<MultiTextDefinition> = [];
+    @Event({ bubbles: true }) valueChanged: EventEmitter<Array<string | MultiTextDefinition>>;
+    @State() currentValues?: Array<MultiTextDefinition> = [];
+    @State() dropdownTags?: Array<MultiTextDefinition> = [];
+
+    @Watch('values')
+    valuesChangedHandler(newValue: Array<string | MultiTextDefinition>) {
+        let values: Array<MultiTextDefinition> = [];
+
+        newValue.forEach(value => {
+            this.dropdownValues.forEach(tag => {
+                if (value === tag.value) {
+                    values.push(tag);
+                }
+            })
+        })
+        this.currentValues = values || [];
+    }
+
+    componentWillLoad() {
+        this.dropdownTags = this.dropdownValues;
+        let values: Array<MultiTextDefinition> = [];
+
+        this.values.forEach(value => {
+            this.dropdownValues.forEach(tag => {
+                if (value === tag.value) {
+                    values.push(tag);
+                }
+            })
+        })
+        this.currentValues = values;
+
+        const tagsToRemove: Array<string> = this.currentValues.map(tag => tag.value);
+        this.dropdownTags = this.dropdownTags.filter(tag => !tagsToRemove.includes(tag.value));
```

Comment:
I didn't know about `includes` - very nice. You could use it instead of my suggestion above where I use `findIndex`.

Reply:
Includes has value as parameter, not a function returning true or false as findIndex does so unsure that they can be interchanged easily.